heartbeat/beater/heartbeat.go

```diff
@@ -336,6 +337,15 @@ func makeESClient(cfg *conf.C, attempts int, wait time.Duration) (*eslegclient.C
 		wait,
 	)
 
+	// Overriding the default ES request timeout:
+	// Higher values of timeouts cannot be applied on the SAAS Service
+	// where we are running in tight loops and want the next successive check to be run for a given monitor
+	// within the next scheduled interval which could be 1m or 3m
+	timeout := int64((10 * time.Second).Seconds())
+	if err := cfg.SetInt("timeout", -1, timeout); err != nil {
```

Comment:
This workds but it should not modify the config in a way that would propagate to the pipeline, it should only apply to the `esloader`:
```
{"log.level":"warn","@timestamp":"2023-11-06T16:26:35.747+0100","log.origin":{"file.name":"beater/heartbeat.go","file.line":92},"message":"Original config list of fields before makeESClient(): [bulk_max_size hosts]","service.name":"heartbeat","ecs.version":"1.6.0"}
...
{"log.level":"warn","@timestamp":"2023-11-06T16:26:44.355+0100","log.origin":{"file.name":"beater/heartbeat.go","file.line":104},"message":"Original config list of fields after makeESClient(): [hosts bulk_max_size timeout]","service.name":"heartbeat","ecs.version":"1.6.0"}
```

Reply:
oh I see what you mean, thanks @emilioalvap, I will check